shopBuddy/shopBuddy/VCs/AddItemViewController.m

```diff
@@ -35,19 +36,35 @@ - (IBAction)didTapScanBarcode:(id)sender {
 }
 
 - (IBAction)didTapSearch:(id)sender {
-    //TODO: Search field validation
+    __weak __typeof__(self) weakSelf = self;
     [[BarcodeAPIManager shared] searchItemsWithQuery:_searchField.text completion:^(NSMutableArray<Item*> *items, NSError *error) {
-        //TODO: handle error
-        self->_searchResults = items;
-        [self->_tableView reloadData];
+        if(items) {
+            __strong __typeof(weakSelf)strongSelf = weakSelf;
+            if(strongSelf)
+            {
+                strongSelf->_searchResults = items;
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [strongSelf->_tableView reloadData];
```

Comment:
This inner block is also async so the strong ref created outside could be nil by the time the block is executed causing a crash. The right way to do it is create another strong ref inside the inner block however for readability purposes is better to move implementation to a private method and call using weakSelf created above.